packages/memory-storage/src/resource-clients/request-queue.ts

```diff
@@ -169,11 +180,22 @@ export class RequestQueueClient extends BaseClient implements storage.RequestQue
 
         const items = [];
 
-        for (const storageEntry of existingQueueById.requests.values()) {
+        const seenRequestIds = new Set<string>();
+        const handledForefrontIds = new Set<string>();
```

Comment:
Could you please comment why you're tracking these?

Reply:
- `seenRequestIds` are tracked, so we don't double-return a single request (both from the forefront list and the original request list)
- `handledForefrontIds` are tracked so we can remove them after the iteration (not during, as that might potentially cause indexing problems, see the discussion at https://github.com/apify/crawlee/pull/2681#discussion_r1778491966 and the [last commit](https://github.com/apify/crawlee/pull/2681/commits/169d48bbfc1763cbbe990f3f64757d8697810d7b).)

Reply:
Ah, uh "comment", as you think this warrants adding an in-code comment, or is this explanation here sufficient?

Reply:
I'd definitely prefer to have it in code :smile: